core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -380,13 +386,19 @@ private BasicInvocationStage handleNonTxWriteCommand(InvocationContext ctx, Writ
          }
          if (!(ce instanceof OutdatedTopologyException) && !(ce instanceof SuspectException))
             throw t;
+         WriteCommand writeCommand = (WriteCommand) rCommand;
+         if (writeCommand.hasFlag(Flag.ZERO_LOCK_ACQUISITION_TIMEOUT)) {
+            // the command is expected to not wait
```

Comment:
You mean that's what `putForExternalRead()` needs, or do you have other use cases in mind?